rustls/src/client/tls13.rs

```diff
@@ -907,11 +907,10 @@ impl State<ClientConnectionData> for ExpectFinished {
             _fin_verified: fin,
         };
 
-        if cx.common.is_quic() {
-            Ok(Box::new(ExpectQuicTraffic(st)))
-        } else {
-            Ok(Box::new(st))
-        }
+        Ok(match cx.common.is_quic() {
```

Comment:
if we use match here, i think doing it on `cx.common.protocol`, `Protocol::Quic`, etc. would be clearer than matching on a bool?

Reply:
(unless we expect to change the detail contained in `is_quic()` in the future, of course)

Reply:
Just because you dislike `match` on `bool`? That doesn't seem like a big deal to me.

IMO `is_quic()` is still a useful shorthand which avoids the need for a `Protocol` import.

Reply:
i think i dislike bool in general and look for any and all opportunities to avoid it; but it's not a big deal :)